packages/error-view/versioned-view.tsx

```diff
@@ -0,0 +1,78 @@
+import React from 'react'
+import { useRouter } from 'next/router'
+import Link from 'next/link'
+import useErrorPageAnalytics from './use-error-page-analytics'
+import ErrorPage, { ErrorPageProps } from '.'
+import s from './style.module.css'
+
+const versionPattern = /\/(?<version>v\d+[.]\d+[.](\d+|x))/
+
+function VersionNotFound({ version }: { version: string }): React.ReactElement {
+  const { asPath } = useRouter()
+  useErrorPageAnalytics(404)
+
+  const pathWithoutVersion = asPath.replace(versionPattern, '')
+  const basePath = asPath.split('/')[1]
+
+  return (
+    <div className={s.root}>
+      <h1 className={s.heading}>
+        This page does not exist for version {version}.
+      </h1>
+      <p>
+        Please select either the{' '}
+        <Link href={pathWithoutVersion}>
+          <a>most recent version</a>
+        </Link>{' '}
+        or a valid version that includes the page you are looking for.
+      </p>
+      <p>
+        <Link href={`/${basePath}`}>
+          <a className={s.link}>← Go back to Documentation</a>
+        </Link>
+      </p>
+    </div>
+  )
+}
+
+/**
+ * A Standalone error page component intended to be used on sites with versioned docs support.
+ * Exports a full, custom implementation of an pages/_error.tsx component.
```

Comment:
I opted to use `pages/_error` here instead of `pages/404` as the page renders differently depending on the URL, and so static generation of the 404 page doesn't really make a lot of sense IMO. The alternative is to SSR it one way and do some client-side munging to render it differently after the initial hydration, at which point why not just SSR?